src/dev/build/tasks/os_packages/docker_generator/run.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { resolve } from 'path';
+import { write, copyAll, exec } from '../../../lib';
+import * as dockerTemplates from './templates';
+
+export async function runDockerGenerator(config, log, build) {
+  const license = build.isOss() ? 'ASL 2.0' : 'Elastic License';
+  const imageFlavor = build.isOss() ? '-oss' : '';
+  const imageTag = 'docker.elastic.co/kibana/kibana';
+  const versionTag = config.getBuildVersion();
+  const artifactsServerContainer = 'kibana-docker-artifacts-server';
```

Comment:
@tylersmalley our build is sequential. First we are running the steps for the `default` and then to the `oss`. Therefore, we are basically generate the build docker scripts with instructions to create the artifacts server using the same, as only one will be running at the same time. Right now, the trap will kill the running server with  `kibana-docker-artifacts-server` name. In the way the code is written, if we use an unique name for each artifacts server, according to the license, each trap would take care only of their own artifact server. Though, I just think we don't need it right now 😃

Reply:
is the artifacts server needed?  when it's in a separate repo it makes sense but if it depends on build artifacts why fetch over curl instead of ADD in the dockerfile?

Reply:
Could we also just add the directory instead of even dealing with the tar?